### PR TITLE
feat: zoom to layer

### PR DESCRIPTION
**Pull request by rot1024**
# Overview
Sometime when we added a layer to the scene, we will lost where they are ( especially the 3D tiles layer ). This feature will trigger the camera fly to the layer which is selected, then users can find where they are.
## What I've done
Add a new button on Layer panel (Left panel ) on the editor , so the user can press on this button and the camera will fly to the Layer.

## Screenshot
image

## Memo
Design Link : https://www.figma.com/file/bdnfDaGXGoNfUws4y7JItl/Re%3AEarth-UI-Gamma?node-id=11570%3A180230

### CODE REVIEW

File: src/components/atoms/Icon/Icons/zoomToLayer.svg [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+<path fill-rule="evenodd" clip-rule="evenodd" d="M1.3335 3.00016C1.3335 2.63197 1.63197 2.3335 2.00016 2.3335H4.50016C4.86835 2.3335 5.16683 2.63197 5.16683 3.00016C5.16683 3.36835 4.86835 3.66683 4.50016 3.66683H2.66683V5.50016C2.66683 5.86835 2.36835 6.16683 2.00016 6.16683C1.63197 6.16683 1.3335 5.86835 1.3335 5.50016V3.00016Z" fill="#C7C5C5"/>
+<path fill-rule="evenodd" clip-rule="evenodd" d="M8.00016 4.8335C8.36835 4.8335 8.66683 5.13197 8.66683 5.50016V10.5002C8.66683 10.8684 8.36835 11.1668 8.00016 11.1668C7.63197 11.1668 7.3335 10.8684 7.3335 10.5002V5.50016C7.3335 5.13197 7.63197 4.8335 8.00016 4.8335Z" fill="#C7C5C5"/>
+<path fill-rule="evenodd" clip-rule="evenodd" d="M4.8335 7.99984C4.8335 7.63165 5.13197 7.33317 5.50016 7.33317L10.5002 7.33317C10.8684 7.33317 11.1668 7.63165 11.1668 7.99984C11.1668 8.36803 10.8684 8.6665 10.5002 8.6665L5.50016 8.6665C5.13197 8.6665 4.8335 8.36803 4.8335 7.99984Z" fill="#C7C5C5"/>
+</svg>
```
Comment (KaWaite) [inline src/components/atoms/Icon/Icons/zoomToLayer.svg:8]: Fills need to be changed to "currentColor"

File: src/components/molecules/EarthEditor/LayerTreeViewItem/LayerActions/index.tsx [hunk 2]
```diff
@@ -66,10 +68,21 @@ const LayerActions: React.FC<Props> = ({
         </HelpButton>
       </Action>
       <Action disabled={!rootLayerId} onClick={importLayer}>
-        <HelpButton descriptionTitle={t("Add Layer.")} balloonDirection="bottom">
+        <HelpButton descriptionTitle={t("Add Layer")} balloonDirection="bottom">
```
Comment (KaWaite) [inline src/components/molecules/EarthEditor/LayerTreeViewItem/LayerActions/index.tsx:71]: Please update all other HelpButton's text to have no period and have each word uppercased

File: src/components/molecules/EarthEditor/OutlinePane/hooks.tsx [hunk 3]
```diff
@@ -384,6 +386,7 @@ export default ({
     visibilityShown: true,
     selectedLayerId,
     rootLayerId,
+    onZoomToLayer: onZoomToLayer,
```
Comment (KaWaite) [inline src/components/molecules/EarthEditor/OutlinePane/hooks.tsx:389]: You only need to do something: something123 if the prop name is different from the function/variable's name

```suggestion
    onZoomToLayer,
```

File: src/components/molecules/EarthEditor/OutlinePane/index.tsx [hunk 4]
```diff
@@ -84,6 +85,7 @@ const OutlinePane: React.FC<Props> = ({
   onLayerImport,
   onLayerGroupCreate,
   onDrop,
+  onZoomToLayer, // TODO
```
Comment (KaWaite) [inline src/components/molecules/EarthEditor/OutlinePane/index.tsx:88]: I'm guessing we don't need the TODO any more :P 
```suggestion
  onZoomToLayer,
```

File: src/components/molecules/Visualizer/Engine/Cesium/useEngineRef.ts [hunk 5]
```diff
@@ -87,6 +87,13 @@ export default function useEngineRef(
           options,
         );
       },
+      lookAtLayer: layerId => {
+        const viewer = cesium.current?.cesiumElement;
+        if (!viewer || viewer.isDestroyed()) return;
+        const e = viewer.entities.getById(layerId);
+        if (!e) return;
+        viewer.flyTo(e);
```
Comment (lavalse) [inline src/components/molecules/Visualizer/Engine/Cesium/useEngineRef.ts:95]: const camera = getCamera(viewer);
        viewer.zoomTo(
          e,
          new Cesium.HeadingPitchRange(camera?.heading ?? 0, camera?.pitch ?? -90, 5000),
        );

Is this a better solution?

**Review by KaWaite — APPROVED** [review]

**Pull request merged**